lib/file.js

```diff
@@ -22,11 +23,15 @@ function File(filename, opts) {
     opts.comment = opts.comment || 'inline';
 
     this._content = [''];
+
     if (opts.sourceMap) {
         this._map = new SourceMapGenerator({
-            file: path.basename(filename),
+            file: opts.sourceMap.from,
```

Comment:
according to JSDoc `sourceMap` should be boolean.
and test for such case is also missing.

And BTW the change looks major. To make it minor let's keep fallback:

```js
opts.sourceMap.from || path.basename(filename)
```

Reply:
@tadatuta 👌